packages/framework-provider-azure-infrastructure/src/infrastructure/synth/terraform-containers.ts

```diff
@@ -1,72 +1,156 @@
-  ): CosmosdbSqlContainer {
+    cosmosdbDatabaseResource: cosmosdbAccount.CosmosdbAccount,
+    cosmosdbSqlDatabaseResource: cosmosdbSqlDatabase.CosmosdbSqlDatabase
+  ): cosmosdbSqlContainer.CosmosdbSqlContainer {
     const readModelName = config.resourceNames.forReadModel(readModel)
     const idReadModel = toTerraformName(readModel)
-    return new CosmosdbSqlContainer(terraformStack, idReadModel, {
+    return new cosmosdbSqlContainer.CosmosdbSqlContainer(terraformStackResource, idReadModel, {
       name: readModelName,
-      resourceGroupName: cosmosdbDatabase.resourceGroupName,
-      accountName: cosmosdbDatabase.name,
-      databaseName: cosmosdbSqlDatabase.name,
+      resourceGroupName: cosmosdbDatabaseResource.resourceGroupName,
+      accountName: cosmosdbDatabaseResource.name,
+      databaseName: cosmosdbSqlDatabaseResource.name,
       partitionKeyPath: '/id',
       partitionKeyVersion: 2,
       autoscaleSettings: {
         maxThroughput: MAX_CONTAINER_THROUGHPUT,
       },
+      provider: providerResource,
+    })
+  }
+
+  private static createSubscriptionsContainer(
+    providerResource: AzurermProvider,
+    appPrefix: string,
+    terraformStackResource: TerraformStack,
+    config: BoosterConfig,
+    cosmosdbDatabaseResource: cosmosdbAccount.CosmosdbAccount,
+    cosmosdbSqlDatabaseResource: cosmosdbSqlDatabase.CosmosdbSqlDatabase
+  ): cosmosdbSqlContainer.CosmosdbSqlContainer {
+    const id = toTerraformName(appPrefix, 'subscriptions-table')
+    return new cosmosdbSqlContainer.CosmosdbSqlContainer(terraformStackResource, id, {
+      name: config.resourceNames.subscriptionsStore,
+      resourceGroupName: cosmosdbDatabaseResource.resourceGroupName,
+      accountName: cosmosdbDatabaseResource.name,
+      databaseName: cosmosdbSqlDatabaseResource.name,
+      partitionKeyPath: `/${subscriptionsStoreAttributes.partitionKey}`,
+      partitionKeyVersion: 2,
+      defaultTtl: -1,
+      autoscaleSettings: {
+        maxThroughput: MAX_CONTAINER_THROUGHPUT,
+      },
+      provider: providerResource,
+    })
+  }
+
+  private static createConnectionsContainer(
+    providerResource: AzurermProvider,
+    appPrefix: string,
+    terraformStackResource: TerraformStack,
+    config: BoosterConfig,
+    cosmosdbDatabaseResource: cosmosdbAccount.CosmosdbAccount,
+    cosmosdbSqlDatabaseResource: cosmosdbSqlDatabase.CosmosdbSqlDatabase
+  ): cosmosdbSqlContainer.CosmosdbSqlContainer {
+    const id = toTerraformName(appPrefix, 'connections-table')
+    return new cosmosdbSqlContainer.CosmosdbSqlContainer(terraformStackResource, id, {
+      name: config.resourceNames.connectionsStore,
+      resourceGroupName: cosmosdbDatabaseResource.resourceGroupName,
+      accountName: cosmosdbDatabaseResource.name,
+      databaseName: cosmosdbSqlDatabaseResource.name,
+      partitionKeyPath: `/${connectionsStoreAttributes.partitionKey}`,
+      partitionKeyVersion: 2,
+      defaultTtl: -1,
```

Comment:
In AWS, the subscriptions and connections table have a field that tells the TTL of the record, so subscriptions are not there forever. Not sure if this is the right configuration field, but, shouldn't we also add this kind of behavior to Azure?

Reply:
Great! thanks for the hint to look for this